packages/backend/src/open_payments/wallet_address/service.ts

```diff
@@ -173,9 +174,14 @@ async function createWalletAddress(
       ? cleanAdditionalProperties(options.additionalProperties)
       : undefined
 
+    const tenantId = options.tenantId
+      ? options.tenantId
+      : deps.config.operatorTenantId
+
     const walletAddress = await WalletAddress.query(
       deps.knex
     ).insertGraphAndFetch({
+      tenantId,
```

Comment:
just to not forget: should check in this fn that the asset is belongs to the correct tenant

Reply:
Done.